servers/visual/shader_language.cpp

```diff
@@ -3373,6 +3373,10 @@ Error ShaderLanguage::_parse_block(BlockNode *p_block, const Map<StringName, Dat
 			cf->blocks.push_back(block);
 			p_block->statements.push_back(cf);
 
+			for (const Map<StringName, BlockNode::Variable>::Element *E = init_block->variables.front(); E; E = E->next()) {
```

Comment:
actually, I think the real problem is that the newly created block a few lines above:
```
BlockNode *block = alloc_node<BlockNode>();
block->parent_block = p_block;
```

should have as parent the init block, this would solve the problem of the variables not being fond.

Reply:
Yeah, what I thought.
Will give it a change as soon as I'm on my study computer :p